chaos_test.go

```diff
@@ -0,0 +1,289 @@
+
+	t.Run("edge case: far future time", func(t *testing.T) {
+		farFuture := time.Now().AddDate(100, 0, 0)
+		result := Time("farFuture")
+		assert.True(t, result.Before(farFuture) || result.Equal(farFuture))
+	})
+}
+
+func TestFloat(t *testing.T) {
+	t.Run("deterministic output", func(t *testing.T) {
+		assert.Equal(t, Float(1.0, "seed"), Float(1.0, "seed"))
+	})
+
+	t.Run("respects upper bound", func(t *testing.T) {
+		for i := 0; i < 1000; i++ {
+			result := Float(10.0, i)
+			assert.GreaterOrEqual(t, result, 0.0)
+			assert.Less(t, result, 10.0)
+		}
+	})
+
+	t.Run("edge case: zero", func(t *testing.T) {
+		assert.Equal(t, 0.0, Float(0.0, "seed"))
+	})
+
+	t.Run("edge case: negative upper bound", func(t *testing.T) {
+		result := Float(-10.0, "seed")
+		assert.GreaterOrEqual(t, result, -10.0)
+		assert.Less(t, result, 0.0)
+	})
+
+	t.Run("edge case: very small positive number", func(t *testing.T) {
+		result := Float(1e-10, "seed")
+		assert.GreaterOrEqual(t, result, 0.0)
+		assert.Less(t, result, 1e-10)
+	})
+}
+
+func TestString(t *testing.T) {
+	t.Run("deterministic output", func(t *testing.T) {
+		assert.Equal(t, String(10, "seed"), String(10, "seed"))
+	})
+
+	t.Run("respects length", func(t *testing.T) {
+		for i := 0; i < 100; i++ {
+			result := String(i, "seed")
+			assert.Len(t, result, i)
+		}
+	})
+
+	t.Run("contains only alphanumeric characters", func(t *testing.T) {
+		result := String(1000, "seed")
+		for _, char := range result {
+			assert.Contains(t, alphanumericalChars, string(char))
+		}
+	})
+
+	t.Run("edge case: empty string", func(t *testing.T) {
+		assert.Equal(t, "", String(0, "seed"))
+	})
+}
+
+func TestSliceItem(t *testing.T) {
+	t.Run("deterministic output", func(t *testing.T) {
+		items := []int{1, 2, 3, 4, 5}
+		assert.Equal(t, SliceItem(items, "seed"), SliceItem(items, "seed"))
+	})
+
+	t.Run("returns zero value for empty slice", func(t *testing.T) {
+		var emptyIntSlice []int
+		assert.Equal(t, 0, SliceItem(emptyIntSlice, "seed"))
+
+		var emptyStringSlice []string
+		assert.Equal(t, "", SliceItem(emptyStringSlice, "seed"))
+	})
+
+	t.Run("selects from all items", func(t *testing.T) {
+		items := []int{1, 2, 3, 4, 5}
+		selected := make(map[int]bool)
+		for i := 0; i < 1000; i++ {
+			selected[SliceItem(items, i)] = true
+		}
+		assert.Len(t, selected, len(items))
+	})
+
+	t.Run("edge case: single item slice", func(t *testing.T) {
+		items := []int{42}
+		assert.Equal(t, 42, SliceItem(items, "seed"))
+	})
+}
+
+func TestUniqueSliceItems(t *testing.T) {
+	t.Run("deterministic output", func(t *testing.T) {
+		items := []int{1, 2, 3, 4, 5}
+		result1, err1 := UniqueSliceItems(items, 3, "seed")
+		result2, err2 := UniqueSliceItems(items, 3, "seed")
+		require.NoError(t, err1)
+		require.NoError(t, err2)
+		assert.Equal(t, result1, result2)
+	})
+
+	t.Run("returns error when not enough items", func(t *testing.T) {
+		items := []int{1, 2, 3}
+		_, err := UniqueSliceItems(items, 4, "seed")
+		assert.Error(t, err)
```

Comment:
`assert.ErrorIs()` ?

Reply:
Using `assert.ErrorIs()` wouldn't be appropriate here because there's no specific error type to compare against.

We could define a custom error type in the chaos package and use that instead of `fmt.Errorf()` 
But I think we just keep it as it is for now.